packages/frontend-app-api/src/wiring/createApp.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createExtension, createPlugin } from '@backstage/frontend-plugin-api';
+import { createInstances } from './createApp';
+
+import { MockConfigApi } from '@backstage/test-utils';
+
+describe('createInstances', () => {
+  it('throws an error when a root extension is overridden', () => {
+    const config = new MockConfigApi({
+      app: {
+        extensions: [
+          {
+            root: {
+              at: '',
+            },
+          },
+        ],
+      },
+    });
+    const plugins = [
+      createPlugin({
+        id: 'plugin',
+        extensions: [],
+      }),
+    ];
+    expect(() => createInstances({ config, plugins })).toThrow(
+      'There is a root extension in the app config file and root extensions are not configurable',
+    );
+  });
+
+  it('throws an error when cyclical dependencies are found', () => {
+    const config = new MockConfigApi({});
+    const plugins = [
+      createPlugin({
+        id: 'plugin',
+        extensions: [
+          createExtension({
+            id: 'root',
```

Comment:
Should this also be validated, to avoid plugin extensions with root ID?

Reply:
Yep this should not be allowed and should result in a simiar error to the above

Reply:
Done: https://github.com/backstage/backstage/pull/20032/commits/2ab65100a554b8d8248d0db187ede4891d9294b7